filebeat/prospector/prospector.go

```diff
@@ -155,6 +158,13 @@ func (p *Prospector) createHarvester(state file.State) (*harvester.Harvester, er
 }
 
 func (p *Prospector) startHarvester(state file.State, offset int64) error {
+
+	if p.config.HarvesterLimit > 0 && atomic.LoadUint64(&p.harvesterCounter) >= p.config.HarvesterLimit {
+		return fmt.Errorf("Harvester limit reached.")
+	}
+
+	atomic.AddUint64(&p.harvesterCounter, 1)
```

Comment:
right. Unfortunately `atomic` does not provide an api to directly get the incremented counter value. Using a mutex for incrementing would be simpler. Otherwise one would need to use CompareAndSwap to atomically update and check counter still being valid.